util/ProtobufUUIDUtil.java

```diff
@@ -6,7 +6,16 @@
 import java.util.UUID;
 
 public class ProtobufUUIDUtil {
+    private enum EmptyUUID {
+        INSTANCE;
+
+        private UUID emptyUuid = new UUID(0, 0);
+    }
+
     public static UUID fromBuf(ByteString uuid) {
+        if (uuid == null || uuid.isEmpty()) {
+            return EmptyUUID.INSTANCE.emptyUuid;
+        }
         ByteBuffer buffer = ByteBuffer.allocate(16).put(uuid.asReadOnlyByteBuffer());
         buffer.rewind();
         long msb = buffer.getLong();
```

Comment:
This looks like another special class that could use some documentation of why it's needed in the comments.